src/blockchain/Blockchain.ts

```diff
@@ -186,6 +174,40 @@ export default class Blockchain implements IBlockchainApi {
     const update = await this.api.tx[this.chainmod].updateAccumulator(
       accumulator.valueOf()
     )
-    await update.signAndSend(address)
+    return new Promise((resolve, reject) => {
+      // store the handle to remove subscription.
+      let unsubscribe: (() => void) | null = null
+
+      // sign and send transaction
+      update
+        .signAndSend(address, (r) => {
```

Comment:
Please add integration tests for this section to get the global coverage for branches above 75% again.

Reply:
I will not add integration tests since this part is already covert with integration tests (each time we update the accumulator and check that the chain has changed. eg. in `is possible to store an accumulator`)

I added some more mocking and this part is know covert by tests.

Rereview? 😋